docs/proposals/deviceapi-edge-design.md

```diff
@@ -0,0 +1,156 @@
+
+Currently, we have defined Device in [device_instance_types.go](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/apis/devices/v1alpha2/device_instance_types.go#L360) and [device.go](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/devicecontroller/types/device.go#L4) respectively, some fields have the same meaning but different names, which is ambiguous. 
+
+When the device status is updated, the "update message" is sent to the edge through the structure defined in  [device.go](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/devicecontroller/types/device.go#L4) . The example mentioned in the background section shows that when device status is updated, we send `DeviceTwinUpdate` to the edge rather than a completed `Device` object, which is different from built-in resource objects (such as Pod). It is suggested to move `DeviceTwinUpdate`, `MembershipUpdate` and other structures in  [device.go](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/devicecontroller/types/device.go#L4) to  [device_instance_types.go](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/apis/devices/v1alpha2/device_instance_types.go#L360) and re-define `DeviceSpec` and `DeviceStatus` , then **we send the whole `Device` object to edge like `Pod` do**. Once we get a whole `Device` on the edge, then we can store device info on the edge and use Device API on the edge. And **this is the key of this proposal**.
+
+* DeviceController
+
+The device controller is the cloud component of KubeEdge which is responsible for device management, and **it is responsible for synchronizing device updates between edge and cloud.** It starts two separate goroutines called `upstream controller` and `downstream controller`(these are not separate controllers as such but named here for clarity). 
+
+Since we modify the `Device` object, then we need to modify device controller(both `downstream` and `upstream` ). The changes are as follows:
+
+* discard configmap when new device created since we store the device properties and visitors etc in device_instance.
+* send a whole `Device` update to edge instead of `membership` update or `twin` update on downstream.
+* send a whole `Device` update to cloud instead of `twin` update on upstream.
+
+At present, device controller follows the design [here](https://github.com/kubeedge/kubeedge/blob/master/docs/components/cloud/device_controller.md#device-controller). 
+
+* DeviceTwin & Mapper
+
+Once we can retrive a whloe `Device` from cloud,  we can simplify the implementation on the edge. For now, we have to defive some types(e.g.  [dttype](https://github.com/kubeedge/kubeedge/tree/master/edge/pkg/devicetwin/dttype) and [mapper type](https://github.com/kubeedge/kubeedge/blob/master/mappers/common/configmaptype.go)) on the edge in order to unmarshal/marshal device info. Once we can use Device API on the edge, it can be optimized.
+
+### Use Cases
+
+* Simplify the implementation for mapper components for user
```

Comment:
- Simplify device API sync logic, improve maintainability
- Simplify the implementation for mapper components for user